chain/chain/src/store.rs

```diff
@@ -2028,6 +2028,43 @@ impl<'a> ChainStoreUpdate<'a> {
         Ok(())
     }
 
+    /// Clears chunk data which can be computed from other data in the storage.
+    ///
+    /// We are storing PartialEncodedChunk objects in the ColPartialChunks in
+    /// the storage.  However, those objects can be computed from data in
+    /// ColChunks and as such are redundant.  For performance reasons we want to
+    /// keep that data when operating at head of the chain but the data can be
+    /// safely removed from archival storage.
+    ///
+    /// `gc_stop_height` indicates height starting from which no data should be
+    /// garbage collected.  Roughly speaking this represents start of the ‘hot’
+    /// data that we want to keep.
+    ///
+    /// `gt_height_limit` indicates limit of how many heights to process.  This
+    /// limit means that the method may stop garbage collection before reaching
+    /// `gc_stop_height`.
+    pub fn clear_redundant_chunk_data(
+        &mut self,
+        gc_stop_height: BlockHeight,
+        gc_height_limit: BlockHeightDelta,
+    ) -> Result<(), Error> {
+        let start = self.chunk_tail()?;
+        let stop = gc_stop_height.min(start.saturating_add(gc_height_limit));
+        for height in start..stop {
```

Comment:
in other GCs - we keep going until we gc exactly 'height_limit' blocks.
Here you  pick 'fixed' limits. (so if there were no blocks produced at this height - it means that we don't GC anything in this round).

Was this intended? if yes - maybe let's quickly mention it in a comment - as this is different behaviour than other GC.

Reply:
I’ve changed it a bit so that we don’t count empty heights. This is more in line with how we treat blocks.